notebooks/peerscout/peerscout-recommend-reviewing-editors.ipynb

```diff
@@ -245,123 +250,105 @@
-    "        'person_id': editor_person_id_by_name_map[predicted_editor[1]],\n",
-    "        'matching_keyword_score': predicted_editor[2],\n",
-    "        'matching_keywords': [{\n",
-    "            'score': keyword_score,\n",
-    "            'keyword': keyword\n",
-    "        } for keyword_score, keyword in predicted_editor[3]],\n",
-    "    }\n",
-    "    for row in prediction_results_df.itertuples()\n",
-    "    for predicted_editor in row.prediction\n",
-    "])\n",
-    "print(\"len of prediction_results_flat_df\", len(prediction_results_flat_df))\n",
-    "prediction_results_flat_df.head(3)"
-   ],
-   "outputs": [],
-   "metadata": {}
+    "prediction_results_df['prediction'][0]"
+   ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "source": [
-    "# prediction_results_flat_df['version_id'].nunique()"
-   ],
+   "metadata": {},
    "outputs": [],
-   "metadata": {}
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
    "source": [
-    "# prediction_results_flat_df.max()"
-   ],
-   "outputs": [],
-   "metadata": {}
+    "print(\"'prediction_results_df' memory usage:\")\n",
+    "prediction_results_df.info(memory_usage='deep')"
+   ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "source": [
-    "prediction_results_flat_df.info(memory_usage='deep')"
-   ],
+   "metadata": {},
    "outputs": [],
-   "metadata": {}
+   "source": [
+    "def iter_prediction_results(\n",
+    "    prediction_df: pd.DataFrame,\n",
+    ") -> Iterable[dict]:\n",
+    "    for row in prediction_df.itertuples():\n",
+    "        for predicted_editor in row.prediction:\n",
+    "            yield {\n",
+    "                'version_id': row.version_id,\n",
+    "                'score': predicted_editor[0],\n",
+    "                'name': predicted_editor[1],\n",
+    "                'person_id': editor_person_id_by_name_map[predicted_editor[1]],\n",
+    "                'matching_keyword_score': predicted_editor[2],\n",
+    "                'matching_keywords': [{\n",
+    "                    'score': keyword_score,\n",
+    "                    'keyword': keyword\n",
+    "                } for keyword_score, keyword in predicted_editor[3]],\n",
+    "            }\n"
+   ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "source": [
-    "print('writing to:', recommendation_output_table_name)\n",
-    "to_gbq(\n",
-    "    prediction_results_flat_df,\n",
-    "    recommendation_output_table_name,\n",
-    "    project_id=project_id,\n",
-    "    if_exists='replace'\n",
-    ")\n",
-    "print('done')"
-   ],
+   "metadata": {},
    "outputs": [],
-   "metadata": {}
+   "source": [
+    "for batch_data_iterable in iter_batches_iterable(iter_prediction_results(prediction_results_df), 10000):\n",
+    "    df_prediction_iterable = pd.DataFrame(batch_data_iterable)\n",
+    "    print('writing to:', recommendation_output_table_name)\n",
+    "    to_gbq(\n",
```

Comment:
Instead of `to_gbq` we could also call `json_list_as_jsonl_file` and `load_file_into_bq_with_auto_schema` directly.
That would work because `json_list_as_jsonl_file` accepts an iterable.
The jsonl file would then contain all of the data.

You may also need to call `remove_key_with_null_value` somewhere. In this case it might be easier to call it inside `iter_prediction_results`.